feature_engine/outliers/winsorizer.py

```diff
@@ -136,108 +246,28 @@ def __init__(
-
-        # find or check for numerical variables
-        self.variables_ = _find_or_check_numerical_variables(X, self.variables)
-
-        if self.missing_values == "raise":
-            # check if dataset contains na
-            _check_contains_na(X, self.variables_)
-            _check_contains_inf(X, self.variables_)
-
-        self.right_tail_caps_ = {}
-        self.left_tail_caps_ = {}
-
-        # estimate the end values
-        if self.tail in ["right", "both"]:
-            if self.capping_method == "gaussian":
-                self.right_tail_caps_ = (
-                    X[self.variables_].mean() + self.fold * X[self.variables_].std()
-                ).to_dict()
-
-            elif self.capping_method == "iqr":
-                IQR = X[self.variables_].quantile(0.75) - X[self.variables_].quantile(
-                    0.25
-                )
-                self.right_tail_caps_ = (
-                    X[self.variables_].quantile(0.75) + (IQR * self.fold)
-                ).to_dict()
-
-            elif self.capping_method == "quantiles":
-                self.right_tail_caps_ = (
-                    X[self.variables_].quantile(1 - self.fold).to_dict()
-                )
-
-        if self.tail in ["left", "both"]:
-            if self.capping_method == "gaussian":
-                self.left_tail_caps_ = (
-                    X[self.variables_].mean() - self.fold * X[self.variables_].std()
-                ).to_dict()
-
-            elif self.capping_method == "iqr":
-                IQR = X[self.variables_].quantile(0.75) - X[self.variables_].quantile(
-                    0.25
-                )
-                self.left_tail_caps_ = (
-                    X[self.variables_].quantile(0.25) - (IQR * self.fold)
-                ).to_dict()
-
-            elif self.capping_method == "quantiles":
-                self.left_tail_caps_ = X[self.variables_].quantile(self.fold).to_dict()
-
-        self.n_features_in_ = X.shape[1]
-
-        return self
+        super().__init__(capping_method, tail, fold, variables, missing_values)
+        self.add_indicators = add_indicators
 
     # Ugly work around to import the docstring for Sphinx, otherwise not necessary
     def transform(self, X: pd.DataFrame) -> pd.DataFrame:
-        X = super().transform(X)
-
-        return X
+        X = _is_dataframe(X)
+        X_out = super().transform(X)
+        if self.add_indicators:
+            if self.tail in ["left", "both"]:
+                X_low = X_out > X
+                X_low.columns = [str(cl) + "_low" for cl in X.columns.values]
+            if self.tail in ["right", "both"]:
+                X_high = X_out < X
+                X_high.columns = [str(cl) + "_high" for cl in X.columns.values]
+            if self.tail in ["left", "both"]:
+                X_out = pd.concat([X_out, X_low], axis=1)
+            if self.tail in ["right", "both"]:
+                X_out = pd.concat([X_out, X_high], axis=1)
+        return X_out
```

Comment:
A few things to consider here:

1. X.columns.values will add indicators to all variables in the data to transform. But the transformer only caps numerical variables or those entered by the user, so here we should probably use `self.variables_`
2. can we change the names low and high to left and right to be consistent with the names we are already using in the docs and in the class?
3. Instead of X_out > X, which will compare the entire dataset and not just the variables of interest, could we add code that is more specific, and make use of the values that were detected per variable during fit and are stored in the dictionaries?